example/src/components/buttons/InCallButton.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { ReactNode } from 'react';
+import { StyleSheet, View, Pressable } from 'react-native';
+
+import { AdditionalColors, BrandColors } from '../../shared/colors';
+
+const IconSize = 25;
+
+const InCallButtonStyles = StyleSheet.create({
+  common: {
+    width: 44,
+    height: 44,
+    borderRadius: 22,
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  primary: {
+    borderWidth: 1,
+    borderColor: BrandColors.darkBlue80,
+    borderStyle: 'solid',
+    backgroundColor: AdditionalColors.white,
+  },
+  danger: {
+    backgroundColor: AdditionalColors.red80,
+  },
+});
+
+type ButtonTypeName = 'primary' | 'danger';
+
+type InCallButtonProps = {
+  type: ButtonTypeName;
+  children: ReactNode;
+};
+
+export const InCallButton = (props: InCallButtonProps) => {
+  const GetStylesForButtonType = (type: ButtonTypeName) => {
+    return [
+      InCallButtonStyles.common,
+      type === 'primary'
+        ? InCallButtonStyles.primary
+        : InCallButtonStyles.danger,
+    ];
+  };
+
+  const GetIconColorForButtonType = (type: ButtonTypeName) => {
+    switch (type) {
+      case 'primary':
+        return BrandColors.darkBlue100;
+      case 'danger':
+        return AdditionalColors.white;
+    }
+  };
+
+  return (
+    <Pressable onPress={() => {}}>
+      <View style={GetStylesForButtonType(props.type)}>
+        {React.cloneElement(props.children as React.ReactElement<any>, {
```

Comment:
Why can't we just pass icon name as a prop and use Icon component here? Wouldn't it be much simpler?

Reply:
That requires developers later on to always maintain the up-to-date list of all possible icons in this file, which I wasn't sure if it's a good practice. If you prefer I can do that.

Reply:
Icon name can be just a string. In fact, the Icon component itself has a prop `name` that's just a string in case of icomoon icons. What a bummer. Maybe there is a clever way to do it, but for now let's make it just a string.

Reply:
Done.